pandas/core/arrays/string_arrow.py

```diff
@@ -416,8 +417,17 @@ def _cmp_method(self, other, op):
         pc_func = ARROW_CMP_FUNCS[op.__name__]
         if isinstance(other, ArrowStringArray):
             result = pc_func(self.data, other.data)
+        elif isinstance(other, np.ndarray):
+            result = pc_func(self.data, other)
         elif is_scalar(other):
-            result = pc_func(self.data, pa.scalar(other))
+            try:
+                result = pc_func(self.data, pa.scalar(other))
+            except (pa.lib.ArrowNotImplementedError, pa.lib.ArrowInvalid):
+                mask = isna(self) | isna(other)
+                valid = ~mask
+                result = np.zeros(len(self), dtype="bool")
+                result[valid] = op(np.array(self)[valid], other)
```

Comment:
Is this needed to do the actual calculation? 
Because if the pyarrow function raises, that basically means that the scalar it either NA or either not a string? In which cases the result is either all NA or all False? 

(but, since this need to be further worked on in general (eg support python-based StringArray, support other comparison ops, etc), it's certainly fine to put this comment on a follow-up TODO list)

Reply:
> In which cases the result is either all NA or all False/all True (depending on the op being eq or ne)?

I started down this route, but it is not all False/all True if there are null values in the array and also needs to put in the true/false logic. It's not difficult, but this instead of writing 'new' logic, this is effectively a copy/paste of code in StringArray that we may be able to refactor out.

Reply:
Yes, let's do this in a targeted follow-up.